lib/controller.rb

```diff
@@ -55,6 +55,19 @@ class SlowFood < Sinatra::Base
     erb :index
   end
 
+  post '/checkout' do
+    @current_order = Order.get(session[:order_id])
+    @total = 0
+    if @current_order.nil?
+      flash[:error] = "Your basket is empty"
+    else
+      @current_order.order_items.each do |order_item|
+        @total = @total + order_item.dish.price
```

Comment:
Imo, the `total` method should be added to the model and not be calculated in the controller. You should be able to access a method `@current_order.total`